src/components/MenuPanel/MenuPanel.js

```diff
@@ -65,11 +65,42 @@ function MenuPanel({
   const [systemAppsOpened, setSystemAppsOpened] = useState(
     systemAppsOpenedState.isOpen()
   )
-  const [systemAppsToggled, setSystemAppsToggled] = useState(false)
+  const [showSystemsAppsAnimation, setShowSystemsAppsAnimation] = useState(
+    false
+  )
+
+  const appGroups = useMemo(
+    () =>
+      appInstanceGroups
+        .filter(appGroup => appGroup.hasWebApp)
+        .map(appGroup => ({
+          ...appGroup,
+          icon: <AppIcon app={appGroup.app} />,
+        })),
+    [appInstanceGroups]
+  )
+
+  const showConsole = consoleVisible || activeInstanceId === 'console'
+  const menuApps = [APP_HOME, appGroups]
+  const systemApps = [
+    APP_PERMISSIONS,
+    APP_APPS_CENTER,
+    APP_ORGANIZATION,
+    ...(showConsole ? [APP_CONSOLE] : []),
+  ]
+
+  const isSystemAppActive = useMemo(
+    () => systemApps.some(systemApp => systemApp.appId === activeInstanceId),
+    [activeInstanceId, systemApps]
+  )
 
   const handleToggleSystemApps = useCallback(() => {
-    setSystemAppsOpened(opened => !opened)
-    setSystemAppsToggled(true)
+    setSystemAppsOpened(opened => {
```

Comment:
This is a bit subjective, but I think it's better to use a side effect in the `setState` updater function to make sure our localstorage value stays in sync with the application state.